tianshou/data/collector.py

```diff
@@ -42,30 +58,56 @@ class CollectStats(CollectStatsBase):
     """The time for collecting transitions."""
     collect_speed: float = 0.0
     """The speed of collecting (env_step per second)."""
-    returns: np.ndarray
+    returns: np.ndarray = field(default_factory=lambda: np.empty(0))
     """The collected episode returns."""
-    returns_stat: SequenceSummaryStats | None  # can be None if no episode ends during collect step
+    returns_stat: SequenceSummaryStats | None = None  # can be None if no episode ends while collecting n_step transitions across all workers
     """Stats of the collected returns."""
-    lens: np.ndarray
+    lens: np.ndarray = field(default_factory=lambda: np.empty(0))
     """The collected episode lengths."""
-    lens_stat: SequenceSummaryStats | None  # can be None if no episode ends during collect step
+    lens_stat: SequenceSummaryStats | None = None  # can be None if no episode ends while collecting n_step transitions across all workers
     """Stats of the collected episode lengths."""
 
+    @classmethod
+    def from_collect_output(
+        cls,
+        episode_count: int,
+        step_count: int,
+        collect_call_duration: float,
+        episode_returns: np.ndarray,
+        episode_lens: np.ndarray,
+    ) -> Self:
+        """Instantiate from variables inside the scope of collector.collect()."""
+        return cls(
+            n_collected_episodes=episode_count,
+            n_collected_steps=step_count,
+            collect_time=collect_call_duration,
+            collect_speed=step_count / collect_call_duration,
+            returns=np.array(episode_returns),
+            returns_stat=SequenceSummaryStats.from_sequence(episode_returns)
```

Comment:
Let's move the declaration of this and lens_stats above the instantiation of the object, a bit easier to read

I'm also wondering whether this could be done in the CollectStatsBase itself in post_init, let's have a look together later